test/client/components/CommunityJoinLinkHandler.test.js

```diff
@@ -0,0 +1,61 @@
+require('../support')
+import { mocks, helpers } from '../../support'
+import CommunityJoinLinkHandler from '../../../src/containers/community/CommunityJoinLinkHandler.js'
+import { renderIntoDocument } from 'react-addons-test-utils'
+import { NAVIGATE } from '../../../src/actions'
+const { createElement } = helpers
+
+describe('CommunityJoinLinkHandler', () => {
+  var store, redirectUrl
+
+  const setup = props => {
+    const component = createElement(CommunityJoinLinkHandler, props, {store})
+    renderIntoDocument(component).getWrappedInstance()
+  }
+
+  beforeEach(() => {
+    store = mocks.redux.store({
+      people: {current: {id: 42}},
+      errors: {}
+    })
+    store.transformAction(NAVIGATE, action => redirectUrl = action.payload)
+  })
+
+  it('redirects to onboarding with new membership', done => {
+    store.getState().people.current.memberships = [{
+      community: {id: '1', name: 'Foomunity', slug: 'foomunity'}
+    }]
+    setup({params: {id: 'foomunity', tagName: 'bar'}})
+    // this is to compensate for the setTimeout in CommunityJoinLinkHandler
+    setTimeout(() => {
```

Comment:
look at helpers.wait in test/support, which is a promisified version of setTimeout -- using it, you can integrate the setTimeout(fn, 10) bit that's repeated in all of these tests into setup, e.g.:

```js
// renamed setup to renderAndWait
renderAndWait({params: {id: 'foomunity', tagName: 'bar'}}, () => {
  expect(redirectUrl).to.equal('/c/foomunity/onboarding')
})
```